pkg/extensions/tbtc/recovery/storage.go

```diff
@@ -147,11 +149,28 @@ func (dis *DerivationIndexStorage) GetNextIndex(extendedPublicKey string) (uint3
 		return 0, err
 	}
 
-	index, err := dis.read(extendedPublicKey)
+	lastIndex, err := dis.read(extendedPublicKey)
 	if err != nil {
 		return 0, err
 	}
-	return uint32(index + 1), nil
+	nextIndex := uint32(lastIndex + 1)
+	numberOfTries := uint32(100)
+	for i := uint32(0); i < numberOfTries; i++ {
+		index := nextIndex + i
+		derivedAddress, err := deriveAddress(strings.TrimSpace(extendedPublicKey), index)
+		if err != nil {
+			return 0, err
+		}
+		ok, err := handle.IsAddressUnused(derivedAddress)
+		err = dis.save(extendedPublicKey, index)
+		if err != nil {
+			return 0, err
+		}
+		if ok {
+			return index, nil
```

Comment:
In relation to one of my previous comments, could we return a derived address here?